configs/recommended.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * @fileoverview the `recommended` config for `eslint.config.js`
+ * @author 唯然<weiran.zsd@outlook.com>
+ */
+
+"use strict"
+
+const { configs, rules } = require("../lib/index.js")
+
+module.exports = {
+    plugins: { n: { rules } },
```

Comment:
This way, each config creates a new plugin object. That might be problematic for users, because the config array [rejects merging](https://github.com/eslint/eslint/blob/559ff4e4bc54a8b6e6b54825d83c532d724204b3/lib/config/flat-config-schema.js#L257-L259) when two config entries specify the same plugin name but with different plugin objects.

For example, I tried this eslint.config.js:

```js
const node = require("eslint-plugin-n");
const nodeRecommended = require("eslint-plugin-n/configs/recommended");

module.exports = [
    nodeRecommended,

    // this could be a loaded shareable config that enables some eslint-plugin-n rules
    {        
        plugins: {
            n: node
        }
    }
];
```

And got `TypeError: Key "plugins": Cannot redefine plugin "n".` when running eslint on a file.

Perhaps `plugins.n` in this plugin's configs could always be just the same `require("../lib/index.js")` object? The extra `configs` key (with the old eslintrc configs) would be ignored.

Reply:
should be addressed in 872bc19efa728df67202702c3d6f26e5f444ea70.